### ISSUE TITLE
Issue with Ag-Grid testing

### ISSUE THREAD

**Issue by spamshaker** [vitest-dev/vitest#3688]
### Describe the bug

module resolution fails while testing.
The `vite` command works fine and the application transpiles properly

### Reproduction

here is an example repo
https://github.com/spamshaker/vitest-ag-grid-issue

npm run dev - works
npm run test - fails before beginning with

SyntaxError: Cannot use import statement outside a module
 ❯ Object.<anonymous> node_modules/ag-grid-react/main.js:1:23

```
Module node_modules/ag-grid-react/lib/agGridReact.js:2 seems to be an ES Module but shipped in a CommonJS package. You might want to create an issue to the package "ag-grid-react" asking them to ship the file in .mjs extension or add "type": "module" in their package.json.

As a temporary workaround you can try to inline the package by updating your config:

// vitest.config.js
export default {
  test: {
    deps: {
      inline: [
        "ag-grid-react"
      ]
    }
  }
}

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

 Test Files  1 failed (1)
      Tests  no tests
   Start at  07:00:14
   Duration  232ms (transform 26ms, setup 0ms, collect 0ms, tests 0ms, environment 63ms, prepare 46ms)
```

### System Info

```shell
System:
    OS: macOS 13.4.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 58.31 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.3.1 - ~/.nvm/versions/node/v20.3.1/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.7.2 - ~/.nvm/versions/node/v20.3.1/bin/npm
    pnpm: 8.6.5 - ~/Library/pnpm/pnpm
  Browsers:
    Chrome: 114.0.5735.133
    Edge: 114.0.1823.58
    Safari: 16.5.1
  npmPackages:
    @vitejs/plugin-react: ^4.0.0 => 4.0.1 
    vite: ^4.3.9 => 4.3.9 
    vitest: ^0.32.2 => 0.32.2
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by smilelk4** [comment]
To add to this, also getting the same with `@ag-grid-community/react@30.0.2`

**Comment by spamshaker** [comment]
@smilelk4 the workaround seems to be working thank you

**Comment by spamshaker** [comment]
Resolved with ag-grid v30.0.3, thank you for quick help with workaround!
